### PR TITLE
Improve storage

### PR DESCRIPTION
**Pull request by networkfusion**
## Description

* Updates name of header
* Changes the drive path to a define
* Changes drive names to indexes for consistancy
* Stops enumerating directory paths when using SPIFFS drives

## Motivation and Context

work towards #2849 

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: src/HAL/Include/nanoHAL_Persistent_Storage.h [hunk 1]
```diff
@@ -47,8 +56,8 @@ typedef enum StorageEventType
 
 // constants to be used throughout the code
 #define SUPPORTED_DRIVES_COUNT (2)
-#define DRIVE_LETTER_LENGTH    sizeof(INDEX0_DRIVE_LETTER)
-#define DRIVE_PATH_LENGTH      sizeof(INDEX0_DRIVE_PATH)
+#define DRIVE_LETTER_LENGTH    (2)
```
Comment (Ellerbach) [inline src/HAL/Include/nanoHAL_Persistent_Storage.h:59]: why changing for 2?

**Review by Ellerbach — COMMENTED** [review]
That look indeed cleaner :-)

File: src/HAL/Include/nanoHAL_Persistent_Storage.h (hunk 1 above)
Comment (networkfusion) [reply src/HAL/Include/nanoHAL_Persistent_Storage.h:59]: My reasoning was that we know the length, so why base it off of a static field that should never change (as a compiler optimization. Happy to revert it though.

**Comment by networkfusion** [comment]
Closing as obsolete.

**Pull request closed without merging**